### PR TITLE
implement syncscopes everywhere

### PR DESCRIPTION
**Pull request by vchuravy**
### TODO
- [x] Tests
- [x] Support in UnsafeAtomicsLLVM ext
- [ ] Tests in UnsafeAtomicsLLVM
- [ ] GPU syncscopes

### CODE REVIEW

File: src/UnsafeAtomics.jl [hunk 1]
```diff
@@ -45,4 +47,8 @@ const seq_cst = Internal.seq_cst
 const acquire_release = acq_rel
 const sequentially_consistent = seq_cst
 
+# SyncScope
+const none = Internal.none
```
Comment (vchuravy) [inline src/UnsafeAtomics.jl:51]: Should we have an alias for this called `system`?

**Comment by vchuravy** [comment]
@maleadt so the last remaining question is what do we do about the GPU backend specific syncscopes.

1. Enumerate all of them like Clang
2. Only enumerate the OpenCL ones so that https://github.com/JuliaGPU/GPUCompiler.jl/pull/652 has something to target

**Comment by maleadt** [comment]
What about leaving it up to the caller and simply using symbols? I'm not sure how we'll be able to generalize this without some input from the back-end anyway (i.e. Metal.jl specifying which syncscope to use for a certain kind of array).

**Comment by vchuravy** [comment]
Yeah, right now, I allow for anything to be wrapped in `LLVMSyncScope`

**Pull request merged**